app/controllers/projects_controller.rb

```diff
@@ -1,7 +1,18 @@
 class ProjectsController < ApplicationController
-  before_action :authenticate_user!, only: %i[new create show]
+  before_action :authenticate_user!, only: %i[new create show index my_projects]
   before_action :set_project, only: [:show]
 
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
```

Comment:
Porque esse Rescue aqui?

Se for só para tratar rotas que não existem, é melhor isso ficar no ApplicationController para o comportamento ser repetido por toda a app. Caso seja algo a mais, daí talvez tenha que pensar se realmente é necessário.

Reply:
Obrigado, retirei o comentário e mudei o método not_found para o ApplicationController.